drafts/pair-programming-mistakes.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Navigating the Challenges of Pair Programming
+subtitle: Lessons from My Journey
+domain: software-engineering-corner.zuehlke.com
+tags: pair-programming, software-development
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1717162463753/6mZkSMtlz.png?auto=format
+publishAs: oliveramstutz
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+Pair programming is a powerful tool in the software development process, but like any tool, it comes with its own set of challenges. From feeling overwhelmed by a partner’s pace to navigating the complexities of remote collaboration, I’ve experienced a range of obstacles in my pair programming journey. But through trial and error, reflection, and a deep dive into the subject, I’ve gained a better understanding of its intricacies. In this blog post, I’ll share some of the mistakes I made along the way and the lessons I learned, so you can avoid these pitfalls and make the most out of your pairing sessions. Whether you’re a seasoned developer or a newcomer to the field, I hope my experiences provide some food for thought. Let’s get started!
+
+
+## Pair programming mistakes I made, so you don't have to
+I recently had the feeling that I'm missing out a lot of potential when pair programming. Identifying the exact issue was not immediate, but there was a clear sense that things could be better. When I dived into the topic, I realized that it's following the [Bushnell's Law](https://en.wikipedia.org/wiki/Bushnell%27s%5FLaw). While it's easy to understand the main concepts like driver - navigator, pair programming is a topic hard to master. If you're interested in the theory like different styles of pair programming, I can highly recommend the article ["on pair programming" from Martin Fowler's blog](https://martinfowler.com/articles/on-pair-programming.html), it provides an excellent entry to the topic. 
+
+As a prelude, I see pair programming as people's business and it is therefore highly subjective. The following examples are based on my own experiences and how I feel about them. 
+
+
+## Overwhelmed by the Driver's Pace
+In the early days of my programmer career, right after university, I found myself working with another newcomer to the team. We were both strangers to each other and the project, which added an extra layer of complexity to our collaboration.
+
+Our roles were clearly defined - he was the driver, I was the navigator. As we ventured into the new project and environment, I found his pace challenging to match. Since he was already a senior, he was swift in his actions, and I often found myself trailing behind, struggling to keep up.
```

Comment:
do we need here the struggling to keep up? Imo and I often found myself trailing behind already says that right?

Reply:
i think it adds a nice touch of flowering. In a technical paper i wouldn't write it that way, but since its a blog I think thats ok :)